docs/content/docs/upgrade_guide.md

```diff
@@ -0,0 +1,114 @@
+---
+title: "Upgrade Guide"
+description: "Upgrade Guide"
+type: docs
+---
+
+# Upgrade Guide
+
+This guide describes how to upgrade the Open Service Mesh (OSM) control plane.
+
+## How upgrades work
+
+OSM's control plane lifecycle is managed by Helm and can be upgraded with [Helm's upgrade functionality](https://helm.sh/docs/intro/using_helm/#helm-upgrade-and-helm-rollback-upgrading-a-release-and-recovering-on-failure), which will patch or replace control plane components as needed based on changed values and resource templates.
+
+### Resource availability during upgrade
+Since upgrades may include redeploying the osm-controller with the new version, there may be some downtime of the controller. While the osm-controller is unavailable, there will be a delay in processing new SMI resources, creating new pods to be injected with a proxy sidecar container will fail, and mTLS certificates will not be rotated.
+
+However, already existing SMI resources will be unaffected (assuming [CRD Upgrades](#CRD-Upgrades) are not needed). This means that the data plane (which includes the Envoy sidecar configs) will also be unaffected by upgrading.
+
+Data plane interruptions are expected if the upgrade includes CRD changes. Streamlining data plane upgrades is being tracked in issue [#512](https://github.com/openservicemesh/osm/issues/512).
+
+## Policy
+
+Only certain upgrade paths are tested and supported.
+
+**Note**: These plans are tentative and subject to change.
+
+Breaking changes in this section refer to incompatible changes to the following user-facing components:
+- `osm` CLI commands, flags, and behavior
+- SMI CRDs and controllers
+
+This implies the following are NOT user-facing and incompatible changes are NOT considered "breaking" as long as the incompatibility is handled by user-facing components:
+- Chart values.yaml
+- `osm-config` ConfigMap
+- Internally-used labels and annotations (monitored-by, injection, metrics, etc.)
+
+Upgrades are only supported between versions that do not include breaking changes, as described below.
+
+For OSM versions `0.y.z`:
+- Breaking changes will not be introduced between `0.y.z` and `0.y.z+1`
+- Breaking changes may be introduced between `0.y.z` and `0.y+1.0`
+
+For OSM versions `x.y.z` where `x >= 1`:
+- Breaking changes will not be introduced between `x.y.z` and `x.y+1.0` or between `x.y.z` and `x.y.z+1`
```

Comment:
I don't think it's being guaranteed any differently than other features. We have tests for it that run in CI and we won't merge changes that fail CI. If we otherwise find it's broken in a particular release, then we'll cut a patch to fix it.

Reply:
I don't think we are there yet, and enforcing it right now could mean a lot of work to support backward compatibility at this moment. Can we pick a release to start this sort of enforcement?

Reply:
I agree we aren't there yet, which is why we aren't planning to enforce it until v1.0.0. I'm guessing most users will expect upgrades to work between minor version bumps after that. There's also the disclaimer at the beginning of this section that says these aren't hard commitments.

Reply:
Thanks for clarifying. If not already mentioned, I'd suggest to add the release starting which this enforcement will happen in the doc.

Reply:
I was hoping that was covered by

> For OSM versions `x.y.z` where `x >= 1`: